checkbox-support/checkbox_support/scripts/run_watcher.py

```diff
@@ -93,124 +90,171 @@ def run(self):
                 print("\n\nINSERT NOW\n\n", flush=True)
             elif self.args.testcase == "removal":
                 print("\n\nREMOVE NOW\n\n", flush=True)
+            print("Timeout: {} seconds".format(ACTION_TIMEOUT), flush=True)
         while p.poll():
             if j.process() != journal.APPEND:
                 continue
-            self._callback([e["MESSAGE"] for e in j if e and "MESSAGE" in e])
+            self._process_lines(
+                [e["MESSAGE"] for e in j if e and "MESSAGE" in e]
+            )
 
-    def _callback(self, lines):
+    def _process_lines(self, lines):
         for line in lines:
             line_str = str(line)
             logger.debug(line_str)
-            self._storage_strategy.do_callback(line_str)
+            self.callback(line_str)
 
-    def _no_storage_timeout(self, signum, frame):
+    def _storage_info_helper(self, reserve, mounted_partition=""):
```

Comment:
ok this is very, very hard to understand. My best understanding is that this function basically dumps some infos in a file so that subsequent jobs can read them and try to use them, so the idea is that this is called, all jobs that test that specific partition are called and then this is called again to remove the infos from the file. 

If this is the correct interpretation consider doing the following:
- split this into two, the `reserve` arg only toggles between storing and removing, you can have two functions, one called `_store_storage_info` and one `_delete_stored_storage_info` 
- update the docstring, I tried several time to read it and it is not clear 
- also update the docstring as it documents a non-existing parameter that is used everywhere in this file (`storage_type`)

Reply:
also, storage type is gone and this is used to store more than one storage type. All of them are called `usb_insert_info`. I've read the TODO but is that something we can just TODO and leave as is?